### PR TITLE
Cgi env builder

### LINKED ISSUE #34: ENV builder
A builder that will make the environmental variables for CGI

### PR DESCRIPTION
**Pull request by pmerku**
An example use is in HTTPResponder.cpp

resolves #34

### CODE REVIEW

File: src/env/ENVBuilder.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+				free(envp[i]);
+			free(envp);
+			ERROR_THROW(MallocError());
+		}
+
+		std::string str = it->first + "=" + it->second;
+		envp[i] = utils::strdup(str);
+
+		i++;
+	}
```
Comment (mrjvs) [inline src/env/ENVBuilder.cpp]: use `new` and `delete` for these pointers. works same as malloc but auto throws when allocation fails
```cpp
char **str = new char*[2]();
  str[0] = new char[3]();
  str[0][0] = 'a';
  str[0][1] = 'b';
  str[0][2] = '\0';
  std::cout << "Hello, " << str[0] << "!\n";
```

File: src/env/env.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+using namespace CGIenv;
+
+env::env() : _envp(NULL) { }
+
+env::~env() {
+	if (_envp) {
+		for (int i = 0; _envp[i]; i++)
+			free(_envp[i]);
+		free(_envp);
```
Comment (mrjvs) [inline src/env/env.cpp]: free using delete (see other review comments)

**Review by mrjvs — CHANGES_REQUESTED** [review]

File: src/env/ENVBuilder.cpp (hunk 1 above)
Comment (pmerku) [reply src/env/ENVBuilder.cpp]: Changed in latest commit

File: src/env/env.cpp (hunk 2 above)
Comment (pmerku) [reply src/env/env.cpp]: changed in latest commit

**Review by mrjvs — APPROVED** [review]

**Review by maartenRamadan — APPROVED** [review]

**Pull request merged**